tests/unit/test_cv_facts_tools.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+from __future__ import (absolute_import, division, print_function)
+import pytest
+import pprint
+from ansible_collections.arista.cvp.plugins.module_utils.facts_tools import CvFactsTools
+# from ansible_collections.arista.cvp.plugins.module_utils.fields import FIELD_FACTS_DEVICE, FIELD_FACTS_CONFIGLET, FIELD_FACTS_CONTAINER
+from tests.lib import mock
+from tests.data import facts_unit
+from tests.lib.parametrize import generate_list_from_dict
+from tests.lib.helpers import setup_custom_logger
+from tests.lib.utils import generate_test_ids_dict, generate_test_ids_list
+
+LOGGER = setup_custom_logger(__name__)
+
+# ---------------------------------------------------------------------------- #
+#   FIXTURES
+# ---------------------------------------------------------------------------- #
+
+@pytest.fixture
+def cvp_database():
+    database = mock.MockCVPDatabase(
+        devices=facts_unit.MOCKDATA_DEVICES,
+        configlets=facts_unit.MOCKDATA_CONFIGLETS,
+        containers=facts_unit.MOCKDATA_CONTAINERS,
+        configlets_mappers=facts_unit.MOCKDATA_CONFIGLET_MAPPERS
+    )
+    yield database
+    LOGGER.debug('Final CVP state: %s', database)
+
+
+@pytest.fixture()
+def fact_unit_tools(request, cvp_database):
+    cvp_client = mock.get_cvp_client(cvp_database)
+    instance = CvFactsTools(cv_connection=cvp_client)
+    LOGGER.debug('Initial CVP state: %s', cvp_database)
+    yield instance
+    LOGGER.debug('Mock calls: %s', pprint.pformat(cvp_client.mock_calls))
+
+
+# ---------------------------------------------------------------------------- #
+#   TEST CASES
+# ---------------------------------------------------------------------------- #
+
+# CvFactsTools.__get_container_name
+
+@pytest.mark.generic
+@pytest.mark.usefixtures("fact_unit_tools")
+@pytest.mark.usefixtures("cvp_database")
+@pytest.mark.parametrize("container_def", generate_list_from_dict(facts_unit.MOCKDATA_CONTAINERS), ids=generate_test_ids_dict)
+def test_CvFactsTools__get_container_name(fact_unit_tools, container_def):
+    result = fact_unit_tools._CvFactsTools__get_container_name(key=container_def[mock.MockCVPDatabase.FIELD_KEY])
+    LOGGER.info('facts_tool response: %s', str(result))
+    assert result == container_def['name']
+    LOGGER.info('Container name for id %s is %s', str(container_def[mock.MockCVPDatabase.FIELD_KEY]), str(result))
+
+# CvFactsTools.__configletIds_to_configletName
+
+@pytest.mark.generic
+@pytest.mark.usefixtures("fact_unit_tools")
+@pytest.mark.usefixtures("cvp_database")
+@pytest.mark.parametrize("configlet_def", facts_unit.MOCKDATA_CONFIGLET_MAPPERS['data']['configlets'], ids=generate_test_ids_list)
```

Comment:
Maybe you wanted to use `generate_test_ids_dict` ?

Reply:
Code updated